src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -52,6 +52,32 @@ FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::~FFN()
       boost::apply_visitor(deleteVisitor));
 }
 
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+void FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::
+  CheckInputDim(size_t numRows, const char *functionName)
+{
+  for (size_t l = 0; l < network.size(); ++l)
+  {
+    size_t layerInSize = boost::apply_visitor(InSizeVisitor(), network[l]);
+    if (layerInSize != 0){
+      if (layerInSize != numRows){
+        std::ostringstream oss;
+        oss << "FNN::" << functionName << ": the first layer of the network "
```

Comment:
Could we use `std::string` for `functionName` instead?  That would match the mlpack codebase better. :+1: